staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -876,6 +903,10 @@ func (v *unstructuredSchemaCoercer) apply(u *unstructured.Unstructured) error {
 		return err
 	}
 
+	if !v.preserveUnknownFields {
+		structuralpruning.Prune(u.UnstructuredContent(), v.structuralSchema)
```

Comment:
In place update is safe because `u.UnstructuredContent()` return the object directly (in case of non-nil). I switched to `u.Object` as we have a concrete unstructured object, not the interface here.

In general, our specification of the `runtime.Unstructured` interface is very weak. Also this special case of `u.UnstructuredContent` returning an empty map if `u.Object` is nil, is questionable. We might want to clarify that some day (not here).

Reply:
> I switched to u.Object as we have a concrete unstructured object, not the interface here.

+1

> In general, our specification of the runtime.Unstructured interface is very weak. Also this special case of u.UnstructuredContent returning an empty map if u.Object is nil, is questionable. We might want to clarify that some day (not here).

+1